Added method for arranging dates by day of week

e-e opened the pull request:
Added method for arranging dates by day of week. Also added example in README. I had a specific use case for this, thought others might find it useful as well.

e-e committed:
bin dates by day of week

e-e committed:
added a couple of tests

e-e committed:
fixed eslint issues

e-e committed:
updated readme with example

jgermade commented:
wow! @e-e sorry for the delay, this days have had being way too intensive!

Let me check the changes in the next days!

jgermade commented on README.md:
````diff
@@ -87,3 +87,78 @@ Month {
+
+> returns:
+
+``` js
````
Hi @e-e regarding this payload, I prefer to use an Array of Arrays, maybe something like getColumns

The resulting payload would be:

> first weekday: 0 (sunday)

``` js
[
  [
    { date: 28, month: 4, year: 2017, previous: true },
    { date: 4, month: 5, year: 2017, current: true },
    { date: 11, month: 5, year: 2017, current: true },
    { date: 18, month: 5, year: 2017, current: true },
    { date: 25, month: 5, year: 2017, current: true },
    { date: 2, month: 6, year: 2017, next: true }
  ],
  [
    { date: 29, month: 4, year: 2017, previous: true },
    { date: 5, month: 5, year: 2017, current: true },
    { date: 12, month: 5, year: 2017, current: true },
    { date: 19, month: 5, year: 2017, current: true },
    { date: 26, month: 5, year: 2017, current: true },
    { date: 3, month: 6, year: 2017, next: true }
  ],
  [
    { date: 30, month: 4, year: 2017, previous: true },
    { date: 6, month: 5, year: 2017, current: true },
    { date: 13, month: 5, year: 2017, current: true },
    { date: 20, month: 5, year: 2017, current: true },
    { date: 27, month: 5, year: 2017, current: true },
    { date: 4, month: 6, year: 2017, next: true }
  ],
  [
    { date: 31, month: 4, year: 2017, previous: true },
    { date: 7, month: 5, year: 2017, current: true },
    { date: 14, month: 5, year: 2017, current: true },
    { date: 21, month: 5, year: 2017, current: true },
    { date: 28, month: 5, year: 2017, current: true },
    { date: 5, month: 6, year: 2017, next: true }
  ],
  [
    { date: 1, month: 5, year: 2017, current: true },
    { date: 8, month: 5, year: 2017, current: true },
    { date: 15, month: 5, year: 2017, current: true },
    { date: 22, month: 5, year: 2017, current: true },
    { date: 29, month: 5, year: 2017, current: true },
    { date: 6, month: 6, year: 2017, next: true }
  ],
  [
    { date: 2, month: 5, year: 2017, current: true },
    { date: 9, month: 5, year: 2017, current: true },
    { date: 16, month: 5, year: 2017, current: true },
    { date: 23, month: 5, year: 2017, current: true },
    { date: 30, month: 5, year: 2017, current: true },
    { date: 7, month: 6, year: 2017, next: true }
  ],
  [
    { date: 3, month: 5, year: 2017, current: true },
    { date: 10, month: 5, year: 2017, current: true },
    { date: 17, month: 5, year: 2017, current: true },
    { date: 24, month: 5, year: 2017, current: true },
    { date: 1, month: 6, year: 2017, next: true },
    { date: 8, month: 6, year: 2017, next: true }
  ]
]
```

> first weekday: 1 (monday)

``` js
[
  [
    { date: 28, month: 4, year: 2017, previous: true },
    { date: 4, month: 5, year: 2017, current: true },
    { date: 11, month: 5, year: 2017, current: true },
    { date: 18, month: 5, year: 2017, current: true },
    { date: 25, month: 5, year: 2017, current: true },
    { date: 2, month: 6, year: 2017, next: true }
  ],
  [
    { date: 29, month: 4, year: 2017, previous: true },
    { date: 5, month: 5, year: 2017, current: true },
    { date: 12, month: 5, year: 2017, current: true },
    { date: 19, month: 5, year: 2017, current: true },
    { date: 26, month: 5, year: 2017, current: true },
    { date: 3, month: 6, year: 2017, next: true }
  ],
  [
    { date: 30, month: 4, year: 2017, previous: true },
    { date: 6, month: 5, year: 2017, current: true },
    { date: 13, month: 5, year: 2017, current: true },
    { date: 20, month: 5, year: 2017, current: true },
    { date: 27, month: 5, year: 2017, current: true },
    { date: 4, month: 6, year: 2017, next: true }
  ],
  [
    { date: 31, month: 4, year: 2017, previous: true },
    { date: 7, month: 5, year: 2017, current: true },
    { date: 14, month: 5, year: 2017, current: true },
    { date: 21, month: 5, year: 2017, current: true },
    { date: 28, month: 5, year: 2017, current: true },
    { date: 5, month: 6, year: 2017, next: true }
  ],
  [
    { date: 1, month: 5, year: 2017, current: true },
    { date: 8, month: 5, year: 2017, current: true },
    { date: 15, month: 5, year: 2017, current: true },
    { date: 22, month: 5, year: 2017, current: true },
    { date: 29, month: 5, year: 2017, current: true },
    { date: 6, month: 6, year: 2017, next: true }
  ],
  [
    { date: 2, month: 5, year: 2017, current: true },
    { date: 9, month: 5, year: 2017, current: true },
    { date: 16, month: 5, year: 2017, current: true },
    { date: 23, month: 5, year: 2017, current: true },
    { date: 30, month: 5, year: 2017, current: true },
    { date: 7, month: 6, year: 2017, next: true }
  ],
  [
    { date: 3, month: 5, year: 2017, current: true },
    { date: 10, month: 5, year: 2017, current: true },
    { date: 17, month: 5, year: 2017, current: true },
    { date: 24, month: 5, year: 2017, current: true },
    { date: 1, month: 6, year: 2017, next: true },
    { date: 8, month: 6, year: 2017, next: true }
  ]
]
```

jgermade commented on month.js:
```diff
@@ -1,14 +1,22 @@
 
+var DAYS = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
```
The goal of previous comment is to avoid use texts in any language (english, spanish or any other)

The idea is to include texts in the repo that includes this one.

jgermade commented on month.js:
```diff
@@ -107,4 +117,13 @@ Month.prototype.next = function () {
 };
 
+Month.prototype.getByDay = function() {
+  var bins = JSON.parse(JSON.stringify(BINS));
```
This method should return original date objects, without copying one by one, just keeping a reference.

jgermade commented:
Hi again @e-e I've added some comments in the code of the PR. Do you want to accomplish this changes?

e-e commented:
@jgermade Sure, I can take a look later tonight

jgermade commented:
@e-e great, thanks!

e-e commented:
@jgermade I can't seem to see your comments. Did you definitely finish the review and submit them?

jgermade commented:
@e-e your're right, I've forgot to submit the review

e-e committed:
changed 'getByDay' method to 'getColumns' #2

this changed what was returned. instead of returning an object keyed by
the string representation of the day of week, now returning an array of
arrays. also removed 'constants' for the bins and days of week, since
those aren't really needed any more.

e-e committed:
Merge branch 'master' into master

e-e committed:
Merge branch 'master' into master

e-e commented:
Hey @jgermade, so if I understood you correctly, the main idea was to change how the data was returned by returning an array of arrays, instead of an object keyed by the English days of the week?

e-e committed:
fixed readme example method name

e-e committed:
Merge branch 'master' of https://github.com/e-e/month

e-e committed:
fixed missing bracket and parens

e-e commented:
@jgermade I'm apologize that this has been such a mess. There was a missing closing bracket and parenthesis on one of your new tests that got missed when I merged your updates right on github. (to clarify, I was the one that messed up the brackets when I did the merge) The travis build failed, but the error message was all but worthless.

e-e committed:
refactor day array initialization

e-e committed:
removed unnecessary commented out code

jgermade commented:
Hi @e-e thanks for your code!

I'm going to create a new release from this point! ( And sorry again for the delay :sweat_smile:  )

e-e commented:
@jgermade no worries! thanks!
